### PR TITLE
Add support for BDD tests (.feature)

### PR DESCRIPTION
**Pull request by edno**
This PR:

- [x] Adds a new feature allowing to support `junit.xml` for [Codeception BDD](https://codeception.com/docs/07-BDD) testsuites
- [x] Covered by tests
- [ ] ~~Doc PR: https://github.com/infection/site/pull/XXX~~

This PR allows running `infection` with test framework `codeception` with Gherkin (BDD) style testsuite where JUnit file reference `.feature` tests instead of classes.

Example of `junit.xml`:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<testsuites>
  <testsuite name="acceptance" tests="4" assertions="11" errors="0" failures="0" skipped="0" time="0.604773">
    <testcase file="/codeception/tests/bdd/FeatureA.feature" name="Feature A: Scenario A1" assertions="6" time="0.039365"/>
    <testcase file="/codeception/tests/bdd/FeatureB.feature" name="Feature B: Scenario B2" assertions="2" time="0.022857"/>
    <testcase file="/codeception/tests/bdd/FeatureA.feature" name="Feature A: Scenario A2" assertions="1" time="0.029354"/>
    <testcase file="/codeception/tests/bdd/FeatureA.feature" name="Feature A: Scenario A3" assertions="2" time="0.025091"/>
  </testsuite>
</testsuites>
```

### CODE REVIEW

File: src/TestFramework/Coverage/XmlReport/JUnitTestFileDataProvider.php [hunk 1]
```diff
@@ -81,6 +81,12 @@ public function getTestFileInfo(string $fullyQualifiedClassName): TestFileTimeDa
             );
         }
 
+        $feature = preg_replace("/^(.*):+.*$/", "$1.feature", $fullyQualifiedClassName);
+        if (!$nodes->length) {
```
Comment (maks-rafalko) [inline src/TestFramework/Coverage/XmlReport/JUnitTestFileDataProvider.php]: could you please use `$nodes->length === 0` as on line 76 to be consistent here. This is also what PHPStan requires - "only booleans in conditions"

Comment (edno) [reply src/TestFramework/Coverage/XmlReport/JUnitTestFileDataProvider.php]: Now fixed ✔️

**Comment by maks-rafalko** [comment]
Please check the [pretty ci build](https://github.com/infection/infection/pull/1033/checks?check_run_id=428510633) as well and run `make cs` locally.

**Comment by edno** [comment]
> * If you want to get it released asap, you can target 0.15 branch and backport changes to the master branch then

I am closing this PR, to be replaced by a PR on 0.15 branch.

File: src/TestFramework/Coverage/XmlReport/JUnitTestFileDataProvider.php [hunk 2]
```diff
@@ -81,6 +81,13 @@ public function getTestFileInfo(string $fullyQualifiedClassName): TestFileTimeDa
             );
         }
 
+        $feature = preg_replace('/^(.*):+.*$/', '$1.feature', $fullyQualifiedClassName);
```
Comment (sanmai) [inline src/TestFramework/Coverage/XmlReport/JUnitTestFileDataProvider.php]: Should this be moved inside the condition below, where this is used?

**Pull request closed without merging**